extensions/amp-analytics/0.1/events.js

```diff
@@ -1213,41 +1220,61 @@ export class VisibilityTracker extends EventTracker {
           reportWhenSpec);
     }
 
+    let unlistenPromise;
     // Root selectors are delegated to analytics roots.
     if (!selector || selector == ':root' || selector == ':host') {
       // When `selector` is specified, we always use "ini-load" signal as
       // a "ready" signal.
-      return visibilityManager.listenRoot(
-          visibilitySpec,
-          this.getReadyPromise(waitForSpec, selector),
-          createReportReadyPromiseFunc,
-          this.onEvent_.bind(
-              this, eventType, listener, this.root.getRootElement()));
+      unlistenPromise = visibilityManagerPromise.then(visibilityManager => {
+        return visibilityManager.listenRoot(
+            visibilitySpec,
+            this.getReadyPromise(waitForSpec, selector),
+            createReportReadyPromiseFunc,
+            this.onEvent_.bind(
+                this, eventType, listener, this.root.getRootElement()));
+      }, () => {});
+    } else {
+      // An AMP-element. Wait for DOM to be fully parsed to avoid
+      // false missed searches.
+      const selectionMethod = config['selectionMethod'] ||
+          visibilitySpec['selectionMethod'];
+      unlistenPromise = this.root.getAmpElement(
+          (context.parentElement || context),
+          selector,
+          selectionMethod
+      ).then(element => {
+        return visibilityManagerPromise.then(visibilityManager => {
+          return visibilityManager.listenElement(
+              element,
+              visibilitySpec,
+              this.getReadyPromise(waitForSpec, selector, element),
+              createReportReadyPromiseFunc,
+              this.onEvent_.bind(this, eventType, listener, element));
+        }, () => {});
+      });
     }
 
-    // An AMP-element. Wait for DOM to be fully parsed to avoid
-    // false missed searches.
-    const selectionMethod = config['selectionMethod'] ||
-          visibilitySpec['selectionMethod'];
-    const unlistenPromise = this.root.getAmpElement(
-        (context.parentElement || context),
-        selector,
-        selectionMethod
-    ).then(element => {
-      return visibilityManager.listenElement(
-          element,
-          visibilitySpec,
-          this.getReadyPromise(waitForSpec, selector, element),
-          createReportReadyPromiseFunc,
-          this.onEvent_.bind(this, eventType, listener, element));
-    });
     return function() {
       unlistenPromise.then(unlisten => {
         unlisten();
       });
     };
   }
 
+  /**
+   * Assert that the setting is measurable with host API
+   * @param {string=} selector
+   * @param {string=} reportWhenSpec
+   */
+  assertMeasurableWithHostApi_(selector, reportWhenSpec) {
+    userAssert(!selector || selector == ':root' || selector == ':host',
+        'Element %s that is not root is not supported with host API',
```

Comment:
We decide to not support element level visibility tracking because 1. there's no request for that
2. When I tested with the mraid API, I found the `visibleRectangle`'s value to not be correct. This can be investigated and fixed later though.